src/toolkit/Community.VisualStudio.Toolkit.Shared/Solution/SolutionItem.cs

```diff
@@ -38,9 +39,18 @@ internal void Update(IVsHierarchyItem item)
         {
             ThreadHelper.ThrowIfNotOnUIThread();
             _item = item;
-            _hierarchy = item.HierarchyIdentity.IsNestedItem ? item.HierarchyIdentity.NestedHierarchy : item.HierarchyIdentity.Hierarchy;
-            _itemId = item.HierarchyIdentity.IsNestedItem ? item.HierarchyIdentity.NestedItemID : item.HierarchyIdentity.ItemID;
+            _hierarchy = item.GetHierarchy();
+            _itemId = item.GetItemId();
             _fullPath = GetFullPath();
+            _isNonVisibleItem = new Lazy<bool>(() =>
+            {
+                if (HierarchyUtilities.TryGetHierarchyProperty(_hierarchy, _itemId, (int)__VSHPROPID.VSHPROPID_IsNonMemberItem, out bool isNonMemberItem)
```

Comment:
This could be reduced to just :
```cs
return HierarchyUtilities.TryGetHierarchyProperty(_hierarchy, _itemId, (int)__VSHPROPID.VSHPROPID_IsNonMemberItem, out bool isNonMemberItem) && isNonMemberItem;
```

Reply:
Definitely better. Thanks!